netmiko/base_connection.py

```diff
@@ -272,6 +295,10 @@ def _read_channel(self):
                     break
         elif self.protocol == 'telnet':
             output = self.remote_conn.read_very_eager().decode('utf-8', 'ignore')
+        elif self.protocol == 'serial':
+            output = ""
+            while (self.remote_conn.in_waiting > 0):
+                output += self.remote_conn.read(self.remote_conn.inWaiting())
```

Comment:
Why call inWaiting() here versus using `in_waiting` that was used in while loop?

http://pyserial.readthedocs.io/en/latest/pyserial_api.html#serial.Serial.in_waiting